base/src/Mp4WriterSink.cpp

```diff
@@ -468,6 +556,15 @@ bool Mp4WriterSink::init()
 			mDetail.reset(new DetailH264(mProp));
 		}
 	}
+	for (auto const& element : inputPinIdMetadataMap)
```

Comment:
why two separate loops ?

Reply:
two loops because - enableMp4Metadata will fail if metadata module is added first in pipeline followed by H264 or Jpeg metadata